### DISCUSSION TITLE
issue: Slow rendering performance with huge input data and validation.

### DISCUSSION THREAD

**Discussion by tamirhal** [react-hook-form/react-hook-form#8327] [Q&A] [answered]
### Version Number

7.28.0

### Codesandbox/Expo snack

https://codesandbox.io/s/bold-worker-5xmpvt?file=/src/App.js

### Steps to reproduce

Hey everyone. 
I'm having an issue while using validation with an input that has a huge amount of data.

So I've used the template, there is an input text and a virtual input ('hugeInput'), looks like when 'firstName' input is changed the 'hugeInput' is being checked and it causes the rendering process to stutter (try to type fast).

I found that if I'm removing the check of the type of the array it fixes the stuttering. also changing the mode to 'onSubmit' or 'onBlur'. but in my case, it needs the 'all' mode. 

Is there a way to optimize it?

### Expected behaviour

Not revalidate unchanged field

### Code of Conduct

- [ ] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
please send a codesandbox so we can reproduce and give you tips on how it can be improved.

**Reply by tamirhal** [reply]
@bluebill1049 A link is already in the topic.

Anyway,
https://codesandbox.io/s/bold-worker-5xmpvt?file=/src/App.js

Thank you!

**Comment by bluebill1049** [comment, marked as answer]
why are you doing this? 

```
const randomArray = Array.from({ length: 30000 }, () =>
  Math.floor(Math.random() * 30000)
);
```

not an issue with hook form with you doing above plus the scheme validation gets run each keystroke.

**Reply by tamirhal** [reply]
It's a mocked data. 

In my case, I have users with a field of checkboxes that has thousands of rows. so I mocked an array with 30 thousand elements to reproduce the issue

**Reply by bluebill1049** [reply]
again thousands rows goes against your schema which is the root cause of the issue, optimize your schema to improve this problem.

**Reply by tamirhal** [reply]
Any suggestion on how to improve the schema in this example?
